contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -859,36 +814,20 @@ handle_where_clause_restargets_right(ParseState *pstate, Node *w_clause, List *e
 
 					if (nodeTag(xpr->rexpr) != T_ColumnRef)
 					{
-						xpr = (A_Expr *) arg;
-
-						if (nodeTag(xpr->rexpr) != T_ColumnRef)
+						if (is_sp_describe_undeclared_parameters)
```

Comment:
does this need to be two separate if statements?

Reply:
Kept it in two separate if intentionally as this is a helper function hence it can be called by some other function. Then when the outer if satisfies, we do not know what will be the behaviour at that time. So, better to put extra check for sp_describe_undeclared_parameters.